### DISCUSSION TITLE
useState in pan gesture callback (onEnd)

### DISCUSSION THREAD

**Discussion by Yudha17** [software-mansion/react-native-gesture-handler#2764] [Q&A] [answered]
Hi, I want to try use pan gesture with add useState in pan gesture callback (onEnd). But the app back into Expo main page. Is it bug or else ?

**Comment by m-bert** [comment, marked as answer]
Hi! If I understood correctly, You want to call `useState` inside `onEnd` callback. However, it seems to go against [rules of hooks](https://react.dev/reference/rules/rules-of-hooks). On web you can see this error message:

<img width="1011" alt="image" src="https://github.com/software-mansion/react-native-gesture-handler/assets/63123542/bf8f9458-2381-48de-8a79-088b16e11d4b">

So to answer your question, no, it is not possible to call `useState` inside callback and it is not a bug.
